### ISSUE TITLE
Problems with CNN model training and prediction for text classification

### ISSUE THREAD

**Issue by dping1** [apache/incubator-mxnet#6780]
Hi,

I was trying to train a text classification model and consistently faced with the following 2 problems. Appreciate any guidance on this.

1) Whenever a dropout layer is added, the kernel would crash when the Module.fit() is called.  If I remove the dropout, the model trains successfully.
2) When trying to make new prediction with the trained model, always receive the following error.
Check failed: shape_[0] >= end (1 vs. 50) Slice end index out of range
For prediction,  I have tried various input data shape combinations, e.g (50, 500), (1, 500) and none seemed to work.  

Please see below for the network definition code:

batch_size = 50
input_x = mx.sym.Variable('data') 
input_y = mx.sym.Variable('softmax_label') 

vocab_size = 10000
sentence_size = 500
num_embed = 300 

embed_layer = mx.sym.Embedding(data=input_x, input_dim=vocab_size, output_dim=num_embed, name='vocab_embed')

conv_input = mx.sym.Reshape(data=embed_layer, target_shape=(batch_size, 1, sentence_size, num_embed))

filter_list=[3, 4, 5] 

num_filter=100
pooled_outputs = []
for i, filter_size in enumerate(filter_list):
    convi = mx.sym.Convolution(data=conv_input, kernel=(filter_size, num_embed), num_filter=num_filter)
    relui = mx.sym.Activation(data=convi, act_type='relu')
    pooli = mx.sym.Pooling(data=relui, pool_type='max', kernel=(sentence_size - filter_size + 1, 1), stride=(1,1))
    pooled_outputs.append(pooli)

# combine all pooled outputs
total_filters = num_filter * len(filter_list)
concat = mx.sym.Concat(*pooled_outputs, dim=1)

h_pool = mx.sym.Reshape(data=concat, target_shape=(batch_size, total_filters))

# dropout layer
dropout=0.0

if dropout > 0.0:
    h_drop = mx.sym.Dropout(data=h_pool, p=dropout)
else:
    h_drop = h_pool
    
# fully connected layer
num_label=2

cls_weight = mx.sym.Variable('cls_weight')
cls_bias = mx.sym.Variable('cls_bias')

fc = mx.sym.FullyConnected(data=h_drop, weight=cls_weight, bias=cls_bias, num_hidden=num_label)

# softmax output
sm = mx.sym.SoftmaxOutput(data=fc, label=input_y, name='softmax')

**Comment by zihaolucky** [comment]
Use ``` to make it easy to read. And paste your error info.

**Comment by formath** [comment]
All fine in my pc. Can you show us more information about your error?
```
MacBook-Pro:cnn_text_classification formath$ python text_cnn.py --num-epochs 5 --batch-size 50 --dropout 0.1
Loading data...
Train/Valid split: 9662/1000
train shape: (9662, 56)
valid shape: (1000, 56)
sentence max words 56
embedding size 300
vocab size 18765
[11:20:47] src/operator/tensor/./matrix_op-inl.h:144: Using target_shape will be deprecated.
[11:20:47] src/operator/tensor/./matrix_op-inl.h:144: Using target_shape will be deprecated.
[11:20:47] src/operator/tensor/./matrix_op-inl.h:144: Using target_shape will be deprecated.
[11:20:47] src/operator/tensor/./matrix_op-inl.h:144: Using target_shape will be deprecated.
INFO:root:Epoch[0] Batch [50]	Speed: 184.72 samples/sec	accuracy=0.532549
INFO:root:Epoch[0] Batch [100]	Speed: 185.66 samples/sec	accuracy=0.610000
INFO:root:Epoch[0] Batch [150]	Speed: 186.49 samples/sec	accuracy=0.610400
INFO:root:Epoch[0] Train-accuracy=0.668372
INFO:root:Epoch[0] Time cost=52.075
INFO:root:Epoch[0] Validation-accuracy=0.627000
INFO:root:Epoch[1] Batch [50]	Speed: 187.13 samples/sec	accuracy=0.767059
INFO:root:Epoch[1] Batch [100]	Speed: 186.27 samples/sec	accuracy=0.804400
INFO:root:Epoch[1] Batch [150]	Speed: 189.12 samples/sec	accuracy=0.803600
INFO:root:Epoch[1] Train-accuracy=0.811163
INFO:root:Epoch[1] Time cost=51.689
INFO:root:Epoch[1] Validation-accuracy=0.709000
INFO:root:Epoch[2] Batch [50]	Speed: 184.56 samples/sec	accuracy=0.867451
INFO:root:Epoch[2] Batch [100]	Speed: 184.62 samples/sec	accuracy=0.883200
INFO:root:Epoch[2] Batch [150]	Speed: 185.72 samples/sec	accuracy=0.890400
INFO:root:Epoch[2] Train-accuracy=0.893953
INFO:root:Epoch[2] Time cost=52.215
INFO:root:Epoch[2] Validation-accuracy=0.754000
INFO:root:Epoch[3] Batch [50]	Speed: 185.52 samples/sec	accuracy=0.926275
INFO:root:Epoch[3] Batch [100]	Speed: 187.23 samples/sec	accuracy=0.938800
INFO:root:Epoch[3] Batch [150]	Speed: 183.87 samples/sec	accuracy=0.934000
INFO:root:Epoch[3] Train-accuracy=0.941860
INFO:root:Epoch[3] Time cost=52.126
INFO:root:Epoch[3] Validation-accuracy=0.764000
INFO:root:Epoch[4] Batch [50]	Speed: 183.35 samples/sec	accuracy=0.959216
INFO:root:Epoch[4] Batch [100]	Speed: 186.36 samples/sec	accuracy=0.968400
INFO:root:Epoch[4] Batch [150]	Speed: 183.62 samples/sec	accuracy=0.966400
INFO:root:Epoch[4] Train-accuracy=0.973488
INFO:root:Epoch[4] Time cost=52.310
INFO:root:Epoch[4] Validation-accuracy=0.781000
```

**Comment by dping1** [comment]
Thanks for you reply.   I was able to resolve problem 2) above.  It was a mistake in my code.

Still have problem with 1) with the dropout.  If I change the dropout value to non-zero, say dropout=0.5, I always get the popup warning below in my Jupyter notebook.  There is no stack trace returned by the Jupyter.  

“The kernel appears to have died. It will restart automatically.”

Any suggestions on how to get kernel crash details?
